### ISSUE TITLE
Defer capturing original matchers

### ISSUE THREAD

**Issue by jcarlson** [PsychoLlama/expect-enzyme#9]
Referring to this code snippet: https://github.com/PsychoLlama/expect-enzyme/blob/master/src/assertions.js#L120-L145

This creates a problem, because definitions for any original matchers are captured _at the time this module is imported or required_ rather than at the time `expect` is extended. This can be problematic when used alongside app-specific matchers with colliding names.

For example, the following does not work:

```javascript
// test/helpers/my-matchers.js
export function toHaveClass(className) { ... }
export function toNotHaveClass(className) { ... }

// test/setup.js
import expect from 'expect'
import myMatchers from './helpers/my-matchers'
import enzymeMatchers from 'expect-enzyme'

expect.extend(myMatchers)
expect.extend(enzymeMatchers)
```

In the above example, as soon as `expect-enzyme` is imported, it captures any existing matchers, but they haven't been extended onto `expect` yet. It is a syntax error to move the `import` statement for `enzyme-expect` lower in the file, as all `import` statements must be at the top.

This can be worked around using `require` syntax as follows:

```javascript
// test/setup.js
import expect from 'expect'

const myMatchers = require('./helpers/my-matchers')
expect.extend(myMatchers)

const enzymeMatchers = require('expect-enzyme')
expect.extend(enzymeMatchers)
```

This second form allows `expect` to be extended with the app-specific matchers before `require`ing the `expect-enzyme` matchers. While this work-around is okay for my current use case, it might be more problematic if another module included `enzyme-expect` before the test setup module.

Just thought you'd like to know.

**Comment by PsychoLlama** [comment]
That's a good point @jcarlson, `import` statements usually go at the top. If people are using multiple extensions for `expect`, the assertion overloading won't work.

What if `expect-enzyme` exported a function instead?

```js
import enzymeMatchers from 'expect-enzyme'
import customMatchers from './helpers/my-matchers'

expect.extend(customMatchers)
expect.extend(enzymeMatchers()) // invocation captures existing assertions
```

It wouldn't be able to see any extensions added afterwards, but it would solve the import dilemma.

**Comment by PsychoLlama** [comment]
I've released v1.0 which exports a function. Instead of capturing the matchers on import, now it waits until it's invoked. That should help keep the `import` statements prettier :wink:
